var/spack/repos/builtin/packages/libmatheval/package.py

```diff
@@ -40,3 +40,6 @@ class Libmatheval(AutotoolsPackage):
 
     # Only needed for unit tests, but configure crashes without it
     depends_on('guile', type='build')
+    patch('guile-2.0.patch', when='^guile@2.0:')
+    if '^guile@2.0:':
+        force_autoreconf = True
```

Comment:
Have you actually tried building this? This syntax doesn't work, a non-empty string always evaluates to True in Python:
```python
>>> bool('Hello World')
True
```
The correct syntax would be:
```python
@property
def force_autoreconf(self):
    return '^guile@2.0:' in self.spec
```

Additionally, `force_autoreconf` requires 4 additional dependencies:

- `m4`
- `libtool`
- `automake`
- `autoconf`

It might be simpler to run `autoreconf` yourself and regenerate the `configure` file. If you're lucky, all you'll need is a patch to `configure` and no additional dependencies. The downside would be needing to do this for each new release. Presumably the `configure.in` file is more stable than the `configure` file.

Reply:
Thanks a lot for your comments. The package did install, since I always needed the patch (guile@1.8 is not available in spack). Of course, my "if'" clause was wrong.
I tried to include the dependencies above, which just revealed the mess with autotools. Using the spack custom-build version of autotools instead of the system-wide installed ones (I'm working on SLES12) just crashed the build process.
So I followed your suggestion patching the configure script, which is hopefully more robust.
Here, I diffed my working ac generated configure script to the original one.
However, the patch file became rather large, since libmatheval is rather old and development does not seem to continue.